controller/metrics/metrics.go

```diff
@@ -160,12 +162,12 @@ func NewMetricsServer(addr string, appLister applister.ApplicationLister, appFil
 
 	mux := http.NewServeMux()
 	registry := NewAppRegistry(appLister, appFilter, appLabels)
-	registry.MustRegister(depth, adds, latency, workDuration, unfinished, longestRunningProcessor, retries)
+
 	mux.Handle(MetricsPath, promhttp.HandlerFor(prometheus.Gatherers{
 		// contains app controller specific metrics
 		registry,
-		// contains process, golang and controller workqueues metrics
-		prometheus.DefaultGatherer,
+		// contains workqueue metrics, process and golang metrics
+		ctrl_metrics.Registry,
```

Comment:
Hey @agaudreault-jive 
I had happened to [work](https://github.com/argoproj/argo-cd/pull/17017) on this issue too and had faced the issue of the Go metrics being registered twice as well :)
So we could probably unregister it first as `metrics.Registry.Unregister(collectors.NewGoCollector())` like [here](https://github.com/argoproj/argo-cd/pull/17017/files#diff-2c2968dfb2d5ad8d8e69946fffc1a6cde0d55d805f9253b0d1a3a6faf1836ad7R168)

Reply:
With the current code, metrics are only registered once. 

The problem was to add a reference to the controller in the unit test to have them there as well.